File: src/main/java/ru/ifmo/is/lab1/adminrequests/dto/AdminRequestDto.java
package ru.ifmo.is.lab1.adminrequests.dto;

import lombok.Data;
import ru.ifmo.is.lab1.adminrequests.Status;
import ru.ifmo.is.lab1.users.User;
import java.time.ZonedDateTime;

@Data
public class AdminRequestDto {
  private int id;
  private User user;
  private Status status;
  private User approvedBy;
  private ZonedDateTime approvalDate;
  private ZonedDateTime createdAt;
}

File: frontend/src/interfaces/dto/adminrequests/AdminRequestDto.ts
import { ZonedDateTime } from 'luxon';

import User from '../../models/User';
import { Status } from '../../models/Status';

export interface AdminRequestDto {
    id: number;
    user: User;
    status: Status;
    approvedBy?: User; // Optional field
    approvalDate?: Date; // Optional field
    createdAt: ZonedDateTime;
}
